target/linux/octeon/base-files/lib/preinit/79_move_config

```diff
@@ -1,21 +1,28 @@
-# Copyright (C) 2014 OpenWrt.org
+#!/bin/sh
 
 move_config() {
 	. /lib/functions.sh
-	. /lib/upgrade/common.sh
-
-	case "$(board_name)" in
+	local board="$(board_name)"
+	local device
+	case "${board}" in
 		erlite)
-			mount -t vfat /dev/sda1 /mnt
-			[ -f "/mnt/$BACKUP_FILE" ] && mv -f "/mnt/$BACKUP_FILE" /
-			umount /mnt
+			device="/dev/sda1"
 			;;
 		itus,shield-router)
-			mount -t vfat /dev/mmcblk1p1 /mnt
-			[ -f "/mnt/$BACKUP_FILE" ] && mv -f "/mnt/$BACKUP_FILE" /
-			umount /mnt
+			device="/dev/mmcblk1p1"
 			;;
 	esac
+
+	# checks for:
+	# 1. device variable is not empty
+	# 2. device variable points to actual block device
+	if [ -n "${device}" ] && [ -b "${device}" ] ; then
+		. /lib/upgrade/common.sh
```

Comment:
What's the benefit of moving this include?

Reply:
it is only needed for `$BACKUP_FILE` variable and we might not even get to the point where it is needed (condition check does not succeed).

if target does not support configuration backup i.e. `generic` target - it won't get to load `/lib/upgrade/common.sh`.

this is the only reason.